src/screenshot.cpp

```diff
@@ -710,13 +710,17 @@ static bool MakeSmallScreenshot(bool crashlog)
 /**
  * Configure a Viewport for rendering (a part of) the map into a screenshot.
  * @param t Screenshot type
+ * @param width the width of the screenshot, or 0 for current viewport width (needs to be 0 with SC_VIEWPORT, SC_CRASHLOG, and SC_WORLD).
+ * @param height the height of the screenshot, or 0 for current viewport height (needs to be 0 with SC_VIEWPORT, SC_CRASHLOG, and SC_WORLD).
  * @param[out] vp Result viewport
  */
-void SetupScreenshotViewport(ScreenshotType t, Viewport *vp)
+void SetupScreenshotViewport(ScreenshotType t, Viewport *vp, uint32 width, uint32 height)
 {
 	switch(t) {
 		case SC_VIEWPORT:
 		case SC_CRASHLOG: {
+			assert(width == 0 && height == 0);
```

Comment:
Assert on user input is not the best thing to do. Especially as asserts can be disabled.

Reply:
But it is not based on user input. It is enforcing a rule described in the doxygen comments.

The only ones calling this function are `MakeLargeWorldScreenshot` and `MakeScreenshotWithConfirm`. The latter always calls it with 0 (and the asserts confirm they do), and the first only allows non-zero with `SC_ZOOMEDIN` and `SC_DEFAULTZOOM`.

This is also the reason I added the asserts, to make sure not someone in a year goes: owh, I want to make the `SC_VIEWPORT` size changeable. He will immediately hit the assert, reminding him that the code is not made to do that.

At least, that is my reasoning :) Open for suggestions and alternatives, of course. But this is not user input :)

Reply:
What did trigger me with this comment, that I can just tell the user when he tries `size` in anything not `normal` or `big` it won't do anything. Might be nicer :D

Reply:
Ah yes, now (on real browser) I see. Reviewing in the mobile app is not the best idea ;)